core/src/test/scala/unit/kafka/utils/MockScheduler.scala

```diff
@@ -96,4 +98,42 @@ case class MockTask(name: String, fun: () => Unit, var nextExecution: Long, peri
     else
       1
   }
+
```

Comment:
If we aren't going to implement the Future stuff, we should just return null for the future, so that tests relying on this behavior break quickly rather than getting weird behavior.  Alternately, it might not be that hard to properly implement it.

Reply:
I tried to return null, but unfortunately it broke tests which is why I half-implemented it here.

Reply:
How did the tests break?

Reply:
Any test that uses a MockTask that closes will try to call cancel on the task. This results in trying to call cancel on a null object.

Reply:
After look at it more, I agree that returning null wouldn't work here.  However, I think returning a useless future that always appears to be successful isn't good.  Why not change the return type to Future rather than ScheduledFuture?  Then the mock classes can return a CompletableFuture which we can complete whenever we need to.

Reply:
Done.

Reply:
To be clear, I don't think `MockTask` should inherit from `CompletableFuture`.  I was just suggesting that we could return a separate future object which could then do what we expected.  I think the minimum here would be cancelling the task when `future.cancel()` was called, blocking util the task was done when `future.get()` was called, etc.

If that seems too hard, we can do it in a follow on change.  If you want to get this in now, you can just go back to returning a fake future that's always completed, and we can tackle the rest in a follow-on change.

Reply:
I'm a bit confused by what you mean by a separate future object that can cancel the task when future.cancel() is called.  Currently, MockScheduler creates MockTasks. Are you saying that in the schedule method, a MockTask should be created and added to the queue and then a separate future should be created that is linked to the MockTask in some way that allows it to remove the task from the queue when future.cancel() is called, etc?

 I didn't implement this functionality because MockTasks are used mostly for testing purposes, and currently there is no test to see if a MockTask can be cancelled. Perhaps we want to do that in the future, but that seems like a lot of extra work for something that is not used currently.